pkg/network/opencontrail/cni/ipc/ipc.go

```diff
@@ -0,0 +1,326 @@
+/*
+Copyright 2016 Juniper Networks, Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package ipc
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/rpc"
+	"os"
+	"strconv"
+	"time"
+)
+
+func makeFileName(dir, vm string) string {
+	return dir + "/label-" + vm
+}
+
+/****************************************************************************
+ * Connection handling
+ ****************************************************************************/
+/* struct to hold data for a connection to VRouter */
+type Connection struct {
+	url    string
+	dir    string // config files are stored here for persistency
+	client *rpc.Client
+}
+
+func (conn *Connection) doOp(op string, msg []byte) (*http.Response, error) {
+	req, err := http.NewRequest(op, conn.url, bytes.NewBuffer(msg))
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	client := new(http.Client)
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Println("JSON op fail")
+		return nil, err
+	}
+
+	log.Println("JSON op successful")
+	return resp, nil
+}
+
+/****************************************************************************
+ * ADD message handling
+ ****************************************************************************/
+// RPC Message definition
+type contrailAddRpc struct {
```

Comment:
I agree on the Init() code changes and renamed RPC to "agent". (FYI, rpc was named to be consistent with vrouter-agent where its called port-rpc.)

Splitting connection to separate file doesnt give much advantage at this stage since there is not much processing there. In future if we end up adding more functionality, then we can split.